cmd/run.go

```diff
@@ -68,9 +138,30 @@ func runFunc(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	if err := run.Run(cmd.Context(), target, source, genVersion, installationURL, repo, repoSubDir, debug); err != nil {
-		return err
+	updatesChannel := make(chan run.UpdateMsg)
+	workflow := run.NewWorkflowStep("Workflow", updatesChannel)
+
+	runFnCli := func() error {
+		l := log.From(cmd.Context()).WithWriter(io.Discard) // Swallow logs other than the workflow display
+		ctx := log.With(cmd.Context(), l)
+		err = run.Run(ctx, target, source, genVersion, installationURL, repo, repoSubDir, debug, workflow)
+
+		if err != nil {
+			workflow.FailWorkflow()
+			updatesChannel <- run.MsgFailed
+
+			return err
+		}
+
+		workflow.SucceedWorkflow()
+		updatesChannel <- run.MsgSucceeded
```

Comment:
nit: but should their be like a finalizeWorkflow function so people don't forget to do this

Reply:
Ah ya that's better 👍